src/Version.h

```diff
@@ -17,26 +18,81 @@
 
 namespace Version {
 
-using exact_width_mumble_raw_version_t = std::uint32_t;
-using mumble_raw_version_t             = unsigned int;
+enum class CompareMode { AtLeast, LessThan };
 
-static_assert(sizeof(exact_width_mumble_raw_version_t) <= sizeof(mumble_raw_version_t),
-			  "Invalid scalar type used to represent Mumble versions");
+//
+// The mumble version format (v2) is a uint64:
+// reserved/unused  major   minor   patch
+// 0xFFFF           0xFFFF  0xFFFF  0xFFFF
+// (big-endian)
+//
+
+using mumble_raw_version_t = std::uint64_t;
+using component_t          = std::uint16_t;
+
+constexpr std::uint64_t OFFSET_UNUSED = 48;
+constexpr std::uint64_t OFFSET_MAJOR  = 32;
+constexpr std::uint64_t OFFSET_MINOR  = 16;
+constexpr std::uint64_t OFFSET_PATCH  = 0;
+constexpr std::uint64_t FIELD_MASK    = 0xFFFF;
+constexpr std::uint64_t FIELD_UNUSED  = (FIELD_MASK << Version::OFFSET_UNUSED);
+constexpr std::uint64_t FIELD_MAJOR   = (FIELD_MASK << Version::OFFSET_MAJOR);
+constexpr std::uint64_t FIELD_MINOR   = (FIELD_MASK << Version::OFFSET_MINOR);
+constexpr std::uint64_t FIELD_PATCH   = (FIELD_MASK << Version::OFFSET_PATCH);
 
 constexpr mumble_raw_version_t UNKNOWN = 0;
 
-unsigned int getRaw(const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+mumble_raw_version_t getRaw(const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
 QString toString(mumble_raw_version_t version);
-bool get(int *major, int *minor, int *patch, const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+bool get(Version::component_t *major, Version::component_t *minor, Version::component_t *patch, const QString &version = QLatin1String(MUMTEXT(MUMBLE_VERSION)));
+
+constexpr Version::component_t getMajor(Version::mumble_raw_version_t version) {
+	return static_cast< Version::component_t >((version & Version::FIELD_MAJOR) >> Version::OFFSET_MAJOR);
+}
+
+constexpr Version::component_t getMinor(Version::mumble_raw_version_t version) {
+	return static_cast< Version::component_t >((version & Version::FIELD_MINOR) >> Version::OFFSET_MINOR);
+}
+
+constexpr Version::component_t getPatch(Version::mumble_raw_version_t version) {
+	return static_cast< Version::component_t >((version & Version::FIELD_PATCH) >> Version::OFFSET_PATCH);
+}
+
+constexpr Version::mumble_raw_version_t toRaw(Version::component_t major, Version::component_t minor, Version::component_t patch) {
+	return ((static_cast< std::uint64_t >(major) << Version::OFFSET_MAJOR)
+			| (static_cast< std::uint64_t >(minor) << Version::OFFSET_MINOR)
+			| (static_cast< std::uint64_t >(patch) << Version::OFFSET_PATCH));
```

Comment:
```suggestion
	return ((static_cast< Version::mumble_raw_version_t >(major) << Version::OFFSET_MAJOR)
			| (static_cast< Version::mumble_raw_version_t >(minor) << Version::OFFSET_MINOR)
			| (static_cast< Version::mumble_raw_version_t >(patch) << Version::OFFSET_PATCH));
```

Reply:
Well then I guess I should set the constexpr FIELD_[...] also to mumble_raw_version_t